client/http/interface.go

```diff
@@ -92,6 +93,8 @@ type Client interface {
 	// Additionally, it is important for the caller to handle the content of the response body properly
 	// in order to ensure that it can be read and marshaled correctly into `res`.
 	WithRespHandler(func(resp *http.Response, res interface{}) error) Client
+	// WithBackoffer sets and returns a new client with the given backoffer.
+	WithBackoffer(*retry.Backoffer) Client
```

Comment:
I prefer caller use `ClientOption` in `NewClient` to set it.

Reply:
Using `WithBackoffer` here allows us to flexibly use different `Backoffer`s for different scenarios and interfaces like:

```go
cli := NewClient()
cli.WithBackoffer(bo1).GetRegion()
cli.WithBackoffer(bo2).GetStore()
```

If we use`ClientOption` for initialization, we may need to create a new `Client` every time we want to use a different `Backoffer`.